pkg/micro-ecc/patches/0002-Include-RIOT-Hardware-RNG-interface.patch

```diff
@@ -1,41 +1,68 @@
-From d2cb5acad29db4c98af1a965a9c0d0ce67729984 Mon Sep 17 00:00:00 2001
-From: Mathias Tausig <mathias.tausig@fh-campuswien.ac.at>
-Date: Mon, 7 Mar 2016 17:20:49 +0100
+From 4347959b1336c33f927211683eb6a8018696f2b0 Mon Sep 17 00:00:00 2001
+From: Wentao Shang <wentaoshang@gmail.com>
+Date: Mon, 12 Dec 2016 16:19:34 -0800
 Subject: [PATCH 2/3] Include RIOT Hardware RNG interface
 
 ---
- uECC.c | 8 +++++++-
- 1 file changed, 7 insertions(+), 1 deletion(-)
+ platform-specific.inc |  4 ++++
+ uECC.c                | 14 ++++++++++++++
+ 2 files changed, 18 insertions(+)
 
+diff --git a/platform-specific.inc b/platform-specific.inc
+index 1bb595a..b13fdbe 100644
+--- a/platform-specific.inc
++++ b/platform-specific.inc
+@@ -5,6 +5,8 @@
+ 
+ #include "types.h"
+ 
++#ifdef FEATURE_PERIPH_HWRNG
++
+ #if (defined(_WIN32) || defined(_WIN64))
+ /* Windows */
+ 
+@@ -64,4 +66,6 @@ static int default_RNG(uint8_t *dest, unsigned size) {
+ 
+ #endif /* platform */
+ 
++#endif /* FEATURE_PERIPH_HWRNG */
++
+ #endif /* _UECC_PLATFORM_SPECIFIC_H_ */
 diff --git a/uECC.c b/uECC.c
-index f65fe37..662d899 100644
+index daa144a..b55afd8 100644
 --- a/uECC.c
 +++ b/uECC.c
-@@ -2,6 +2,7 @@
+@@ -2,6 +2,9 @@
  
  #include "uECC.h"
  #include "uECC_vli.h"
++#ifdef FEATURE_PERIPH_HWRNG
 +#include "periph/hwrng.h"
++#endif
  
  #ifndef uECC_RNG_MAX_TRIES
      #define uECC_RNG_MAX_TRIES 64
-@@ -170,10 +171,15 @@ static cmpresult_t uECC_vli_cmp_unsafe(const uECC_word_t *left,
+@@ -181,9 +184,20 @@ static cmpresult_t uECC_vli_cmp_unsafe(const uECC_word_t *left,
      #include "asm_avr.inc"
  #endif
  
++#ifdef FEATURE_PERIPH_HWRNG
 +int riot_hwrng(uint8_t *dest, unsigned size) {
-+  hwrng_read(dest, size);
-+  return 1;
++    hwrng_read(dest, size);
++    return 1;
 +}
++#endif
 +
++#ifdef FEATURE_PERIPH_HWRNG
  #if default_RNG_defined
  static uECC_RNG_Function g_rng_function = &default_RNG;
  #else
--static uECC_RNG_Function g_rng_function = 0;
 +static uECC_RNG_Function g_rng_function = &riot_hwrng;
++#endif
++#else
+ static uECC_RNG_Function g_rng_function = 0;
```

Comment:
I followed your suggestion to split the test into two parts. Now both are passing the CI build. Please take another look.